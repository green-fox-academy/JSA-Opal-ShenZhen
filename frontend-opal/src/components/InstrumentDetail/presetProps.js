import PropTypes from 'prop-types';

const defaultDetailData = [];
const defaultChartData = [];
const detailDataTypes = PropTypes.arrayOf(
  PropTypes.shape({
    id: PropTypes.number,
    name: PropTypes.string,
    value: PropTypes.string,
    group: PropTypes.number
  })
);
const chartDataTypes = PropTypes.arrayOf(
  PropTypes.shape({
    x: PropTypes.string,
    y: PropTypes.number
  })
);

export default {
  defaultDetailData,
  defaultChartData,
  detailDataTypes,
  chartDataTypes
};
