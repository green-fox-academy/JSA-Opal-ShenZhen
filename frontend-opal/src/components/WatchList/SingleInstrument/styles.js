import { StyleSheet } from 'react-native';

import tools from 'components/common/tools';

export default StyleSheet.create({
  container: {
    marginLeft: tools.getPixel(20),
    marginTop: tools.getPixel(20),
    marginRight: tools.getPixel(10)
  }
});
