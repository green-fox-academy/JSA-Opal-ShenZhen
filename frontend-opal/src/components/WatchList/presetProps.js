import PropTypes from 'prop-types';

const defaultQuote = {
  symbol: '',
  latestPrice: 0,
  changePercent: '',
  marketCap: '',
  week52High: 0,
  week52Low: 0,
  peRatio: 0,
  eps: '',
  divYield: '',
  chartData: []
};

const quoteTypes = PropTypes.shape({
  symbol: PropTypes.string,
  latestPrice: PropTypes.number,
  changePercent: PropTypes.string,
  marketCap: PropTypes.string,
  week52High: PropTypes.number,
  week52Low: PropTypes.number,
  peRatio: PropTypes.number,
  eps: PropTypes.string,
  divYield: PropTypes.string,
  chartData: PropTypes.arrayOf(
    PropTypes.shape({
      x: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
      y: PropTypes.number
    })
  )
});

export default {
  defaultQuote,
  quoteTypes
};
