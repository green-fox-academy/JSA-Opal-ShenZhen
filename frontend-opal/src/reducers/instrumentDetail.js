const pairs = {
  GET_INSTRUMENT_DETAIL_DATA: (state, action) => action.payload.data,
  CLEAR_DETAIL_DATA: (state, action) => []
};

export default pairs;
