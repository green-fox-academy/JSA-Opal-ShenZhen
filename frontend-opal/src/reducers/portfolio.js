const pairs = {
  GET_PORTFOLIO_DATA: (state, action) => {
    return [...state, action.payload.data];
  }
};

export default pairs;
