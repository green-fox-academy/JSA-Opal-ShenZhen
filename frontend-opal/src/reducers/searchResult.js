const pairs = new Map();
pairs.set('SEARCH', search);

function searchResult(state = {}, action) {
  if (pairs.has(action.type)) return pairs.get(action.type)(state, action);
  return state;
}

function search(state, action) {
  return action.payload.results;
}

export default searchResult;
