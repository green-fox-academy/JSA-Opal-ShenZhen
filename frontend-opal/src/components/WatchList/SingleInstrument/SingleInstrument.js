import React from 'react';
import { TouchableOpacity } from 'react-native';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import InstrumentRow from 'components/WatchList/InstrumentRow/InstrumentRow';
import ChartDetail from 'components/WatchList/ChartDetail/ChartDetail';
import presetProps from 'components/WatchList/presetProps';
import actions from 'actions/watchlists';

function SingleInstrument({ data, index, toggleOneInstrument }) {
  return (
    <>
      <TouchableOpacity onPress={() => toggleOneInstrument(index, data.symbol)}>
        <InstrumentRow data={data} />
      </TouchableOpacity>
      {data.foldStatus && <ChartDetail data={data} />}
    </>
  );
}

SingleInstrument.defaultProps = {
  data: presetProps.defaultQuote,
  index: 0,
  toggleOneInstrument: null
};

SingleInstrument.propTypes = {
  data: presetProps.quoteTypes,
  index: PropTypes.number,
  toggleOneInstrument: PropTypes.func
};

const mapDispatchToProps = dispatch => ({
  toggleOneInstrument: (index, symbol) => dispatch(actions.toggleOneInstrument(index, symbol))
});

export default connect(null, mapDispatchToProps)(SingleInstrument);
