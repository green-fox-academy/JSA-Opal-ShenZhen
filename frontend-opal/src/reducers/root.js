import { combineReducers } from 'redux';

import watchLists from './Watchlists';
import searchResult from './SearchResult';

export default combineReducers({
  watchLists,
  searchResult,
});
