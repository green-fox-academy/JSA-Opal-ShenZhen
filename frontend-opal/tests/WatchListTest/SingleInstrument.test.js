import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { createStore, compose, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';

import reducers from 'reducers/root';
import SingleInstrument from 'components/WatchList/SingleInstrument/SingleInstrument';
import presetProps from 'components/WatchList/presetProps';

describe('<SingleInstrument />', () => {
  it('renders correctly', () => {
    const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
    const store = createStore(reducers, composeEnhancers(applyMiddleware(thunk)));
    const tree = renderer
      .create(
        <Provider store={store}>
          <SingleInstrument data={presetProps.defaultQuote} index={0} toggleOneInstrument={null} />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
