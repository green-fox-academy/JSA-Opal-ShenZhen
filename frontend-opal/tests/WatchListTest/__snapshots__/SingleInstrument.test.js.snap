// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SingleInstrument /> renders correctly 1`] = `
<View
  style={
    Object {
      "marginLeft": 40,
      "marginRight": 20,
      "marginTop": 40,
    }
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "display": "flex",
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "width": "20%",
          }
        }
      >
        <Text
          style={Object {}}
        >
          
        </Text>
      </View>
      <View
        style={
          Object {
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "width": "80%",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "flex-end",
              "display": "flex",
              "width": "20%",
            }
          }
        >
          <Text
            style={Object {}}
          >
            0
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "flex-end",
              "display": "flex",
              "width": "20%",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "green",
              }
            }
          >
            %
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "flex-end",
              "display": "flex",
              "width": "20%",
            }
          }
        >
          <Text
            style={Object {}}
          >
            
          </Text>
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "flex-end",
              "display": "flex",
              "justifyContent": "center",
              "opacity": 1,
              "width": "20%",
            }
          }
        >
          <Image
            source={1}
            style={
              Object {
                "height": 14,
                "width": 14,
              }
            }
          />
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "flex-end",
              "display": "flex",
              "justifyContent": "center",
              "opacity": 1,
              "width": "20%",
            }
          }
        >
          <Image
            source={1}
            style={
              Object {
                "height": 14,
                "width": 14,
              }
            }
          />
        </View>
      </View>
    </View>
  </View>
</View>
`;
